pkg/scheduler/util/non_zero.go

```diff
@@ -72,13 +72,10 @@ func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.Resourc
 		}
 		return quantity.Value()
 	default:
-		if IsScalarResourceName(resource) {
-			quantity, found := (*requests)[resource]
-			if !found {
-				return 0
-			}
-			return quantity.Value()
+		quantity, found := (*requests)[resource]
```

Comment:
can you please check if resource names declared by the node get validated by the api-server?

Reply:
Are you referring to default resource names, or something else?

Reply:
`ValidateNode` calls [ValidateNodeResources](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L4732) which checks if node resources have nonnegative value and if the resource is integer (hardcoded list or when `IsExtendedResourceName` is true) it checks if the value is still valid.

So no, node's resource names are not validated.

Reply:
@ahg-g even when `GetNonzeroRequestForResource` gets invoked over node's resources, it's still up to the caller to check if the resource name is valid. `GetNonzeroRequestForResource` itself does not need to know if the resource getting retrieved from pod's containers is valid. It's either there or it's not.

Reply:
btw, I feel more comfortable if we checked at the beginning of the function that requests is not nil and return 0 if true.

Reply:
Done